### PR TITLE
feat: improving unit tests for Basic BLS and enforcing Basic BLS for Evo Nodes

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
https://github.com/dashpay/dash/issues/5471

## What was done?
It splits from https://github.com/dashpay/dash/pull/5443

Adds extra unit tests for BLS basic scheme; enforces BLS basic for Evo Nodes in serialization/unserialization of CProRegTx, CProUpServTx.

## How Has This Been Tested?
Run functional/unit tests + added new unit tests.

## Breaking Changes
Serialization slightly changed, but it should be not breaking change

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by ogabrielides — CHANGES_REQUESTED** [review]
Could add the same check on `CProRegTx`.

**Review by ogabrielides — APPROVED** [review]
utACK

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
we should really have similar checks in corresponding `IsTriviallyValid`s too

**Review by ogabrielides — DISMISSED** [review]
utACK

**Comment by knst** [comment]
> we should really have similar checks in corresponding IsTriviallyValids too

We indeed need to have this validation, I added them in PR. Btw, there's no unit tests for it because:
```
    //TODO: Provide raw data for basic scheme as well
```
I created issue for it: https://github.com/dashpay/dash/issues/5471

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/evo/providertx.h [hunk 1]
```diff
@@ -71,7 +77,7 @@ class CProRegTx
                 obj.scriptPayout,
                 obj.inputsHash
         );
-        if (obj.nVersion == BASIC_BLS_VERSION && obj.nType == MnType::HighPerformance) {
+        if (obj.nType == MnType::HighPerformance) {
```
[the remark this answers is not in the document]
Comment (knst) [reply src/evo/providertx.h:80]: agree, make sense, the error message would be more readable.

**Review by ogabrielides — APPROVED** [review]
ACK :+1:

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by ogabrielides** [comment]
@PastaPastaPasta Reasons for keeping this one open?

**Comment by PastaPastaPasta** [comment]
It seems like we're not gonna merge this for v20 right? I think that's better but that's correct right?

**Comment by knst** [comment]
> @[knst](https://github.com/knst) knst modified the milestones: [20](https://github.com/dashpay/dash/milestone/34), [21](https://github.com/dashpay/dash/milestone/38) [Oct 23, 2023](https://github.com/dashpay/dash/pull/5463#event-10742875205) last week

> It seems like we're not gonna merge this for v20 right? I think that's better but that's correct right?

I intentionally changed milestone to v21 - we have already RC; let's don't increase scope for v20.

**Comment by UdjinM6** [comment]
Hmm... I would actually suggest to merge it into v20 - it streamlines the logic and extends unit tests a lot. Imo it a valuable one.

File: src/evo/providertx.h [hunk 2]
```diff
@@ -59,6 +59,7 @@ class CProRegTx
             // unknown version, bail out early
             return;
         }
+
```
Comment (PastaPastaPasta) [inline src/evo/providertx.h:62]: nit unrelated

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**